sourcegraph/sourcegraph
# (chore) rename API repo key-value-pair API fields to repo metadata

Part of https://github.com/sourcegraph/pr-faqs/issues/96.

This PR renames (adds new and deprecates old to be specific) GQL schema:
- `addRepoKeyValuePair` to `addRepoMetadata`
- `updateRepoKeyValuePair` to `updateRepoMetadata`
- `deleteRepoKeyValuePair` to `deleteRepoMetadata`
- `repo.keyValuePairs` to `repo.metadata`
- refactors code accordingly

Note: behind API on the backend it is still called `repo_kvps`

## Test plan
- `sg start`
- Enable the `repository-metadata` feature flag
- CRUD some key-value pairs through API GQL mutation `{add|update|delete}RepoMetadata` API
- Search with `select:repo` and check that repo metadata is present on repo match results as well as repo root pages
